# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../dietlibc/termios.h-ppc-alpha.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

The some constants for the termios functionality are different on powerpc and
alpha. Only tested on powerpc (with bash and pdksh), though.

  - Rene Rebe <rene@exactcode.de>

--- dietlibc/include/termios.h	2005-08-03 08:31:57.000000000 +0200
+++ dietlibc/include/termios.h	2005-08-03 08:33:22.000000000 +0200
@@ -325,6 +325,23 @@
 #define CRTSCTS	  020000000000		/* flow control */
 
 /* c_lflag bits */
+#if defined (__powerpc__) || defined (__alpha__)
+#define ISIG    0x00000080
+#define ICANON  0x00000100
+#define XCASE   0x00004000
+#define ECHO    0x00000008
+#define ECHOE   0x00000002
+#define ECHOK   0x00000004
+#define ECHONL  0x00000010
+#define NOFLSH  0x80000000
+#define TOSTOP  0x00400000
+#define ECHOCTL 0x00000040
+#define ECHOPRT 0x00000020
+#define ECHOKE  0x00000001
+#define FLUSHO  0x00800000
+#define IEXTEN  0x00000400
+#define PENDIN  0x20000000
+#else
 #define ISIG	0000001
 #define ICANON	0000002
 #define XCASE	0000004
@@ -346,6 +363,7 @@
 #define IEXTEN	0100000
 #endif
 #define PENDIN	0040000
+#endif
 
 /* tcflow() and TCXONC use these */
 #define TCOOFF		0
